The package cannot be built
```
Run cd packages/vyos-1x; dpkg-buildpackage -uc -us -tc -b
dpkg-buildpackage: info: source package vyos-1x
dpkg-buildpackage: info: source version 1.5dev0
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by VyOS maintainers and contributors <maintainers@vyos.io>
 dpkg-source --before-build .
dpkg-buildpackage: info: host architecture amd64
dpkg-checkbuilddeps: error: Unmet build dependencies: python3-nose
dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting
dpkg-buildpackage: warning: (Use -d flag to override.)
Error: Process completed with exit code 3.
Run cd packages/vyos-1x; dpkg-buildpackage -uc -us -tc -b
dpkg-buildpackage: info: source package vyos-1x
dpkg-buildpackage: info: source version 1.5dev0
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by VyOS maintainers and contributors <maintainers@vyos.io>
 dpkg-source --before-build .
dpkg-buildpackage: info: host architecture amd64
dpkg-checkbuilddeps: error: Unmet build dependencies: python3-nose
```

Most likely, you have added/copied the old code from the repo instead of adding the newest with only those changes.
Rebase, please, and force push (or clone from the upstream the current codebase)
```
git remote add upstream https://github.com/vyos/vyos-1x.git
git fetch upstream current
git checkout T7723
git rebase upstream/current
git push --force-with-lease
```

--- a/interface-definitions/include/dhcp/option-v4.xml.i
+++ b/interface-definitions/include/dhcp/option-v4.xml.i
@@ -83,6 +83,18 @@
         </constraint>
       </properties>
     </leafNode>
+    <leafNode name="interface-mtu">
+      <properties>
+        <help>Interface MTU</help>
+        <valueHelp>
+          <format>u16:576-9000</format>
+          <description>Client interface MTU</description>
+        </valueHelp>
+        <constraint>
+          <validator name="numeric" argument="--range 576-9000"/>
+        </constraint>
+      </properties>
+    </leafNode>
     <leafNode name="ip-forwarding">
       <properties>
         <help>Enable IP forwarding on client</help>

--- a/python/vyos/kea.py
+++ b/python/vyos/kea.py
@@ -45,6 +45,7 @@ kea4_options = {
     'ipv6_only_preferred': 'v6-only-preferred',
     'captive_portal': 'v4-captive-portal',
     'capwap_controller': 'capwap-ac-v4',
+    'interface_mtu': 'interface-mtu',
 }
 
 kea6_options = {

--- a/smoketest/scripts/cli/test_service_dhcp-server.py
+++ b/smoketest/scripts/cli/test_service_dhcp-server.py
@@ -224,6 +224,7 @@ class TestServiceDHCPServer(VyOSUnitTestSHIM.TestCase):
         server_identifier = bootfile_server
         ipv6_only_preferred = '300'
         capwap_access_controller = '192.168.2.125'
+        interface_mtu = '1420'
 
         pool = base_path + ['shared-network-name', shared_net_name, 'subnet', subnet]
         self.cli_set(pool + ['subnet-id', '1'])
@@ -232,6 +233,7 @@ class TestServiceDHCPServer(VyOSUnitTestSHIM.TestCase):
         self.cli_set(pool + ['option', 'name-server', dns_1])
         self.cli_set(pool + ['option', 'name-server', dns_2])
         self.cli_set(pool + ['option', 'domain-name', domain_name])
+        self.cli_set(pool + ['option', 'interface-mtu', interface_mtu])
         self.cli_set(pool + ['option', 'ip-forwarding'])
         self.cli_set(pool + ['option', 'smtp-server', smtp_server])
         self.cli_set(pool + ['option', 'pop-server', smtp_server])
@@ -363,6 +365,11 @@ class TestServiceDHCPServer(VyOSUnitTestSHIM.TestCase):
             obj,
             ['Dhcp4', 'shared-networks', 0, 'subnet4', 0, 'option-data'],
             {'name': 'ip-forwarding', 'data': 'true'},
+        )
+        self.verify_config_object(
+            obj,
+            ['Dhcp4', 'shared-networks', 0, 'subnet4', 0, 'option-data'],
+            {'name': 'interface-mtu', 'data': interface_mtu},
         )
 
         # Time zone

